code/modules/cargo/blackmarket/blackmarket_market.dm

```diff
@@ -5,13 +5,34 @@
 	/// Available shipping methods and prices, just leave the shipping method out that you don't want to have.
 	var/list/shipping
 
+	/// Amount of time before the market is repopulated
+	var/time_left = 0
+
+	/// Amount of time that time_left is set to after market is repopulated
+	var/max_time_left = 0
 
 	// Automatic vars, do not touch these.
 	/// Items available from this market, populated by SSblackmarket on initialization.
 	var/list/available_items = list()
 	/// Item categories available from this market, only items which are in these categories can be gotten from this market.
 	var/list/categories	= list()
 
+/datum/blackmarket_market/New()
+	. = ..()
+	if(max_time_left)
+		START_PROCESSING(SSprocessing,src)
+
+/datum/blackmarket_market/Destroy(force, ...)
+	if(max_time_left)
+		STOP_PROCESSING(SSprocessing,src)
+	return ..()
+
+/datum/blackmarket_market/process(delta_time)
+	time_left -= delta_time
+	if(time_left <= 0)
+		time_left = max_time_left
+		SSblackmarket.repopulate_market(type)
+
```

Comment:
why use processing and not just... a timer?

Reply:
because of time display, it is easier to display a counting down variable in TGUI than a timer.